packages/legend-graph/src/models/metamodels/pure/valueSpecification/INTERNAL__PropagatedValue.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright (c) 2020-present, Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  ValueSpecification,
+  type ValueSpecificationVisitor,
+} from './ValueSpecification';
+
+export class INTERNAL__PropagatedValue extends ValueSpecification {
+  isDefaultDatePropagationSupported!: boolean;
```

Comment:
After the discussion, I think we can safely remove this flag now. We can check if we can `pop` or `remove` the propagated value based on its context rather than some flags embedded in it. This will keep this type of value spec generic. 

Now in the case of milestoning, when we cleanup the bubbles as we build the value spec from the state, we can have a centralized logic to determine if we can pop that bubble, and the logic here follows the chart you kindly provided. The logic within that function is something we will focus on reviewing.